GoogleForCreators/web-stories-wp
# Editor: Changed video recording trim to happen before insertion

## Context

This moves the actual trimming of the video recording up so it happens as soon as the user presses "trim", and not only when they press "insert".

## Todo

- [x] Make it work!
- [x] Fix bug around re-trimming ~(partially fixed, see comment below)~
- [x] Fix UX around waiting for trim to complete (with a progress bar perhaps?)

## User-facing changes
This flow now happens when trimming a new recording - the "cancel trim" button in the footer of course also works.
![trimmy](https://user-images.githubusercontent.com/637548/182708433-c4a55d34-e60c-4135-b38b-0ebb37494f3b.gif)

## Testing Instructions

Steps to validate:
1. Start a video recording and record 10+ seconds of video
2. Click the trim button to start trimming the recording
3. Move any trim offset and press trim
4. Observe that trim happens instantly with a dialog informing you of this
5. Wait until trim completes (the shorter the faster)
6. Observe that the displayed video is only the trimmed bit
7. Click the trim button again
8. Observe that the original video is available to trim from, with the previously trimmed bit "preselected"
9. Abort trimming
10. Observe that the original trimmed video still displays correctly
11. Press insert
12. Observe that the trimmed video is uploaded and inserted to the page
13. Observe that the trimmed video has the correct duration and thumbnail

## Checklist




---

Fixes #11988
